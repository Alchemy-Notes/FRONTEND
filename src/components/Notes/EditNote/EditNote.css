.form {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.input {
  display: flex;
  margin-bottom: 1rem;
}

.editNoteWrapper {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.markdownGuideWrapper {
  display: flex;
  flex-direction: row;
}

.markdownGuideOpen {
  display: block;
  height: auto;
  max-width: 250px;
  margin: 20px;
  padding: 5px;
  background-color: #000;
  color: #fbf5d6;
  font-size: smaller;
}

.markdownGuideClosed {
  display: none;
}

table,
td,
th {
  border: #fbf5d6 1px solid;
  border-collapse: collapse;
  padding: 2px;
}
