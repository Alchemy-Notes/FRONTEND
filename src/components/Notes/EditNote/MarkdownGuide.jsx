import styles from './EditNote.css';

export default function MarkdownGuide({ toggle }) {
  return (
    <div
      className={toggle ? styles.markdownGuideOpen : styles.markdownGuideClosed}
    >
      <h3>Markdown Guide</h3>
      <table className={styles.markdownTab}>
        <thead>
          <tr>
            <th>Element</th>
            <th>Syntax</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>code block</td>
            <td>
              ```
              <p /> Text <p /> ```
            </td>
          </tr>
          <tr>
            <td>H1</td>
            <td># Text</td>
          </tr>
          <tr>
            <td>H2</td>
            <td>## Text</td>
          </tr>
          <tr>
            <td>H6</td>
            <td>###### Text</td>
          </tr>
          <tr>
            <td>Italic</td>
            <td>*Text*</td>
          </tr>
          <tr>
            <td>Bold</td>
            <td>**Text**</td>
          </tr>
          <tr>
            <td>Strikethrough</td>
            <td>~~Text~~</td>
          </tr>
          <tr>
            <td>Subscript</td>
            <td>Text~sub</td>
          </tr>
          <tr>
            <td>Superscript</td>
            <td>Text^super</td>
          </tr>
          <tr>
            <td>Blockquote</td>
            <td>&gt; Text</td>
          </tr>
          <tr>
            <td>Unordered List</td>
            <td>
              - Item 1 <p />- Item 2
            </td>
          </tr>
          <tr>
            <td>Ordered List</td>
            <td>
              1. Item 1 <p />
              2. Item 2
            </td>
          </tr>
          <tr>
            <td>Code</td>
            <td>`Text`</td>
          </tr>
          <tr>
            <td>Horizontal Rule</td>
            <td>---</td>
          </tr>
          <tr>
            <td>Link</td>
            <td>[text](www.text.com)</td>
          </tr>
          <tr>
            <td>Image</td>
            <td>![alt text](image.png)</td>
          </tr>
          <tr>
            <td>Table</td>
            <td>
              | Text | Text | <p />
              |--|--| <p />
              | Other | Text | <p />
              |More | Text |
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  );
}
