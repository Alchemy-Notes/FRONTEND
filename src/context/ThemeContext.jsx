import { createContext, useContext, useState } from 'react';

const ThemeContext = createContext();

const ThemeProvider = ({ children }) => {
  const [theme, setTheme] = useState(false);

  return (
    <ThemeContext.Provider value={{ theme, setTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};

const useTheme = () => {
  const context = useContext(ThemeContext);

  if (context === undefined) {
    throw new Error('useTheme must be used within a ThemeContext Provider');
  }

  return context;
};

export { ThemeProvider, useTheme };
