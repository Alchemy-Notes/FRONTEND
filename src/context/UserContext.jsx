import { createContext, useContext, useEffect, useState } from 'react';
import { getMe } from '../services/auth';
import { getUserNotes } from '../services/notes';

const UserContext = createContext();

const UserProvider = ({ children }) => {
  const [user, setUser] = useState(null);
  const [notes, setNotes] = useState(null);
  const [tree, setTree] = useState(null);

  useEffect(() => {
    async function get() {
      const result = await getMe();
      setUser(result);
    }
    get();
  }, []);

  useEffect(() => {
    async function getNotes() {
      const result = await getUserNotes({
        userId: user.id,
        query: { type: 'favorite' },
      });
      setNotes(result);
    }
    if (user && user.id) getNotes();
  }, [user]);

  return user && notes ? (
    <UserContext.Provider
      value={{ user, setUser, notes, setNotes, tree, setTree }}
    >
      {children}
    </UserContext.Provider>
  ) : (
    <></>
  );
};

const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('useUser must be used within a UserContext Provider');
  }

  return context;
};

export { UserProvider, useUser };
