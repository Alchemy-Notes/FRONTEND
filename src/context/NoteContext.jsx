import { createContext, useContext, useEffect, useState } from 'react';
import { getUserNotes } from '../services/notes';
import { useUser } from './UserContext';

const NoteContext = createContext();

const NoteProvider = ({ children }) => {
  const [notes, setNotes] = useState([]);
  const { user } = useUser();

  useEffect(() => {
    async function get() {
      const result = await getUserNotes({
        userId: user.id,
        query: { type: 'favorite' },
      });
      setNotes(result);
    }
    if (user && user.id) get();
  }, []);

  return (
    <NoteContext.Provider value={{ notes, setNotes }}>
      {children}
    </NoteContext.Provider>
  );
};

const useNotes = () => {
  const context = useContext(NoteContext);

  if (context === undefined) {
    throw new Error('useNotes must be used within a NoteContext Provider');
  }

  return context;
};

export { NoteProvider, useNotes };
