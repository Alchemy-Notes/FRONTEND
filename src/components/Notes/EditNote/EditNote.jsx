import styles from './EditNote.css';
import { useState } from 'react';
import { useParams, useHistory } from 'react-router-dom';
import AddTags from '../AddTags/AddTags';
import { useUser } from '../../../context/UserContext';
import { addNote, updateNote } from '../../../services/notes';
import Button from '../../Button/Button';

export default function EditNote({ isEditing = false }) {
  const history = useHistory();
  const { noteId } = useParams();
  const { notes, setNotes, user } = useUser();
  const { title, body, prevTags } = notes.filter((note) => note.id === noteId);
  const noteData = isEditing ? { title, body } : { title: '', body: '' };
  const [formState, setFormState] = useState(noteData);
  const [tags, setTags] = useState(prevTags || []);

  const handleSubmit = async (e) => {
    e.preventDefault();
    let response;
    if (isEditing) {
      response = await updateNote(
        { ...formState, tags, userId: user.id },
        noteId
      );
    } else {
      response = await addNote({ ...formState, tags, userId: user.id });
    }
    setNotes((prevState) => [...prevState, response]);
    history.replace(`/notes/${response.id}`);
  };

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormState((prevState) => ({ ...prevState, [name]: value }));
  };

  const handleTagChange = (e) => {
    setTags((prevState) => [...prevState, e.target.value]);
  };

  return (
    <>
      <h1>Edit a note here</h1>
      <form className={styles.form} onSubmit={handleSubmit}>
        <label htmlFor="title">Title: </label>
        <input
          id="title"
          name="title"
          type="text"
          value={formState.title}
          onChange={handleChange}
        />
        <label htmlFor="body">Body:</label>
        <textarea
          id="body"
          name="body"
          rows={6}
          cols={80}
          value={formState.body}
          onChange={handleChange}
        />
        {/* probably need tags to be it's own form */}

        <Button buttonText="Submit" />
      </form>
      <AddTags />
    </>
  );
}
