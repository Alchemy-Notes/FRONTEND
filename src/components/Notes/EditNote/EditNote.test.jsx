import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom/';
import { UserProvider } from '../../../context/UserContext';
import EditNote from './EditNote';

it('should render our button to log into github', () => {
  const { container } = render(
    <MemoryRouter>
      <UserProvider>
        <EditNote />
      </UserProvider>
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
