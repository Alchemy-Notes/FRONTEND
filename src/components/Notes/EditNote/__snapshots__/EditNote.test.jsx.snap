// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render our EditNote component 1`] = `
<div>
  <h1>
    Edit a note here
  </h1>
  <form
    class="form"
  >
    <label
      for="title"
    >
      Title: 
    </label>
    <input
      id="title"
      name="title"
      type="text"
      value=""
    />
    <label
      for="body"
    >
      Body:
    </label>
    <textarea
      cols="80"
      id="body"
      name="body"
      rows="6"
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
  <section>
    <div>
       
    </div>
    <form>
      <label
        for="tags"
      >
        Tags 
      </label>
      <input
        id="tags"
        value=""
      />
      <button
        type="submit"
      >
        Add Tag
      </button>
    </form>
  </section>
</div>
`;
